buildvertical.targets

```diff
@@ -8,6 +8,22 @@
  <!-- Required for FindBestConfiguration task -->
  <Import Project="$(MSBuildThisFileDirectory)src/Tools/GenerateProps/properties.props" />
 
+ <Target Name="AnnotateProjects"
+         BeforeTargets="BuildAllProjects"
+         Condition="!$(MSBuildProjectFullPath.EndsWith('.proj'))">
+   <!-- Clear ProjectWithConfiguration to prevent circular dependency -->
+   <ItemGroup><ProjectWithConfiguration Remove="@(ProjectWithConfiguration)" /></ItemGroup>
```

Comment:
Yes, that's the reason.  We have to mutate the Project item group and trim unapplicable projects for a particular configuration before we can do the traversal